### ISSUE TITLE
IPackageInstallerService queries the source for NuGet, for what purpose? 

### ISSUE THREAD

**Issue by nkolev92** [dotnet/roslyn#35404]
Follow up to https://github.com/dotnet/roslyn/pull/35364. 

The service that recommends packages based on the APIs used queries nuget for the sources enabled? 

What's the purpose of this? 
Does it query the sources on its own, or is it used to simply pass them back to nuget when calling the IVSPackageInstaller/IVSPackageInstaller2 APIs. 

@sharwell

**Comment by sharwell** [comment]
@CyrusNajmabadi Can you give an overview of what we use the `PackageSources` property for?

**Comment by jmarolf** [comment]
So we have an index of all the types in the to 1,000 most downloaded nuget packages on nuget.org.  Theoretically when the feature was written we thought about including other nuget sources or allowing users to index their own but that isn't done yet.

My understanding is that we only wanted to offer to add nuget packages if:
1. The user is actually using nuget.org to get their packages (several repos in dotnet for example use no packages from nuget.org and get them all from private feeds)
2. We were able to download the index file that contains the mapping from type names -> nuget packages

cc: @CyrusNajmabadi to correct anything I got wrong
